builtin/providers/aws/resource_aws_emr.go

```diff
@@ -0,0 +1,439 @@
+package aws
+
+import (
+	"log"
+
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/terraform/helper/schema"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+func resourceAwsEMR() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEMRCreate,
+		Read:   resourceAwsEMRRead,
+		Update: resourceAwsEMRUpdate,
+		Delete: resourceAwsEMRDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"release_label": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"master_instance_type": &schema.Schema{
```

Comment:
I see `master_instance_type` and a few other attributes are not updatable it seems; there's no code in `UPDATE` to change them. Is this intentional? I have not used EMR so maybe this is normal? 

If these are attributes that you cannot update on a _live_ EMR cluster, then they should be marked as `ForceNew` so that changing them re-creates the cluster